### DISCUSSION TITLE
How to solve "mbrola: Warning: p-o unknown, replaced with _-_"?

### DISCUSSION THREAD

**Discussion by GithubAnon0000** [espeak-ng/espeak-ng#2003] [Q&A] [answered]
I am running `espeak-ng` with the voice mb-de7 and want to say the german word "Hypothese" (h,ypot'e:z@). It works fine with `espeak-ng -v German "Hypothese" -x` but not with `espeak-ng -v mb-de7 "Hypothese" -x`. I get the following output with mb-de7 voice:
```
h,ypot'e:z@
mbrola: Warning: p-o unknown, replaced with _-_
mbrola: Warning: o-t unknown, replaced with _-_
```

I found an old and solved issue here that basically said "update espeak-ng to the latest version" and that fixed it for this person. So I compiled it according to the [build instructions](https://github.com/espeak-ng/espeak-ng/blob/master/docs/building.md#linux-mac-bsd). But the same thing happened again (now with eSpeak NG text-to-speech: 1.52-dev).

I assume I have to edit and compile some file in the `phsource/` folder (and not the dict files in `dictsource/`). I'm lost here though.

Any help is appreciated!

**Comment by hozosch** [comment, marked as answer]
It's the "o" phoneme that's the problem, it should be "o:" or "O". Personaly, I think I prefer "o:". Back in the days of original ESpeak, when Espeakedit was still around, you could trace the rules used to create the output. Can you do that with ESpeak-ng as well somehow? Otherwise, it may be a good idea to search de_list or de_rules for things like "hypo" or something. It should be changed there.

**Reply by GithubAnon0000** [reply]
Thanks! I now changed [de_list](https://github.com/espeak-ng/espeak-ng/blob/master/dictsource/de_list) like so (line 778 to 781):
```
hypothese	hypo:'te:z@
hypothenuse	hypo:'te'nu:z@	$4
hypothenusen	hypo:'te'nu:zn-	$4
hypothenusenabschitt	hypo:'te'nu:zn-,?apSnIt	$4
```
and it works fine now with mbrola too. Much appreciated!

Should I convert this discussion to an issue (or open a pr) for that?


**Edit:**
> Back in the days of original ESpeak, when Espeakedit was still around, you could trace the rules used to create the output. Can you do that with ESpeak-ng as well somehow?

I don't know. But `espeakedit` is still in debian oldstable. Not very reliable though, crashed quite often for me and lots of ui issues (e.g. fonts way too small to be readable, barely 2pt in size if at all).

**Reply by GithubAnon0000** [reply]
btw. how did you know that the "o" phoneme was the problem?

**Comment by hozosch** [comment]
Don't know whether this needs to be converted to an issue, a PR may be a good idea tough. I guessed that the "o" was the problem, because I know that this isn't a normal part of ESpeak's German phoneme inventory. ESpeak has it, but it's not normal for German usually. All the other words have really weird phoneme transcriptions as well which you have corrected. They don't seem to affect MBrola, but in all of the Hypothenuse stuff, "e" should be "e:". And there's a Typo in "Hypothenusenabschnitt", the "n" in Abschnitt is missing in the normally written word, not the phoneme transcription.
